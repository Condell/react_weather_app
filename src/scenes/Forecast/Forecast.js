import React from "react";
import styled from "styled-components";

const ForecastWrapper = styled.div`
  background-color: blue;
  font-size: 1.5em;
`;

const Forecast = () =>
  <ForecastWrapper>
    <h1>
      Forecast Component
    </h1>
  </ForecastWrapper>;

export default Forecast;
