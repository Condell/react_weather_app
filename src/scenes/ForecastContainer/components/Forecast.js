import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const ForecastWrapper = styled.div`
  background-color: blue;
  font-size: 1.5em;
`;

const Forecast = props =>
  <ForecastWrapper>
    {props.isLoading === true
      ? <h1>Loading...</h1>
      : <h1>Forecast Component</h1>}
  </ForecastWrapper>;

Forecast.propTypes = {
  isLoading: PropTypes.bool.isRequired
};

export default Forecast;
