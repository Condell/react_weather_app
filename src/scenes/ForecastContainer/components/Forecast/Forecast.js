import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import DaysList from "./components/DaysList/DaysList";
import Title from "../../../../components/Title/Title";

const ForecastWrapper = styled.div`background-color: white;`;

const Forecast = props =>
  <ForecastWrapper {...props}>
    {props.isLoading === true
      ? <h1>Loading...</h1>
      : <DaysList
          name={props.weatherData.city.name}
          data={props.weatherData}
        />}
  </ForecastWrapper>;

Forecast.propTypes = {
  weatherData: PropTypes.object
};



export default Forecast;
