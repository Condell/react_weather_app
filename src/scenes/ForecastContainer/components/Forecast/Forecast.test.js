import React from "react";
import { shallow } from "enzyme";
import 'jest-styled-components'
import toJson from "enzyme-to-json"

import Forecast from "./Forecast";

jest.mock("./Forecast.js", () => "Forecast")

describe("Forecast", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Forecast />);
  });

  it("should render", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});