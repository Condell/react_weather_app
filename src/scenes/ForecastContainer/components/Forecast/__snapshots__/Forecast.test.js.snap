// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast should render 1`] = `<Forecast />`;
